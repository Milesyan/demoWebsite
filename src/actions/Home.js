import { ActionTypes } from './ActionTypes';

export function setHomeStatusPreview() {
  return {
    type: ActionTypes.SET_HOME_STATUS_PREVIEW,
  };
}

export function setHomeStatusProcess() {
  return {
    type: ActionTypes.SET_HOME_STATUS_PROCESS,
  };
}
