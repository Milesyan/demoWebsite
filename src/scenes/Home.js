import React, { PureComponent } from 'react';
import { Header, Footer, HomeContent } from '../components';
import styles from './Home.module.scss';

export default class Home extends PureComponent {
  render() {
    return (
      <div className={styles.container}>
        <Header/>
        <HomeContent/>
        <Footer/>
      </div>
    )
  }
}