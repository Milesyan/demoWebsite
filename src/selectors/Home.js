export const getHomeStatus = state => state.Home.status;

export const getShowLoginDialogStatus = state => state.Home.showLoginDialog;