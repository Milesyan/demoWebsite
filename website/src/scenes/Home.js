import React, { PureComponent } from 'react';
import { Header, Footer, HomeContent } from '../components';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import styles from './Home.module.scss';
class Home extends PureComponent {

  render() {
    return (
      <div className={styles.container}>
        <Header/>
        <HomeContent/>
        <Footer/>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  userInfo: state.userInfo
})

const mapDispatchToProps = dispatch => bindActionCreators({
  // submitUserInfo: 
}, dispatch);
export default connect(mapStateToProps, mapDispatchToProps)(Home);