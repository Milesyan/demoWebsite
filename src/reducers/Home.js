import { ActionTypes } from '../actions/ActionTypes';
export const HOME_STATUS = {
  initial: Symbol(),
  preview: Symbol(),
  process: Symbol(),
  done: Symbol()
}
const defaultState = {
  status: HOME_STATUS.initial
}
const Home = (state = defaultState, action) => {
  switch (action.type) {
    case ActionTypes.SET_HOME_STATUS_INITIAL:
      return { status: HOME_STATUS.initial };
    case ActionTypes.SET_HOME_STATUS_PREVIEW:
      return { status: HOME_STATUS.preview };
    case ActionTypes.SET_HOME_STATUS_PROCESS:
      return { status: HOME_STATUS.process };
    case ActionTypes.SET_HOME_STATUS_DONE:
      return { status: HOME_STATUS.done };
    default:
      return state;
  }
}

export default Home;