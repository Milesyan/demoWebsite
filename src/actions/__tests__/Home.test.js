import * as actions from '../Home'

describe('Home actions', () => {
  
  it('should match snapshot for pure object actions', () => {
    expect(actions.showRequestInviteDialog()).toMatchSnapshot();
    expect(actions.dismissRequestInviteDialog()).toMatchSnapshot();
  });

});
