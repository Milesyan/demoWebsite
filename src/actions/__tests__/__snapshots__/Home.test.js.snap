// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home actions should match snapshot for pure object actions 1`] = `
Object {
  "type": "HOME_SHOW_REQUEST_INVITE_DIALOG",
}
`;

exports[`Home actions should match snapshot for pure object actions 2`] = `
Object {
  "type": "HOME_DISMISS_REQUEST_INVITE_DIALOG",
}
`;
